nflow-engine/src/main/java/io/nflow/engine/internal/dao/WorkflowInstanceDao.java

```diff
@@ -354,10 +354,16 @@ public void updateWorkflowInstanceAfterExecution(WorkflowInstance instance, Work
   public int updateWorkflowInstance(WorkflowInstance instance) {
     // using sqlVariants.nextActivationUpdate() requires that nextActivation is used 3 times
     Object nextActivation = sqlVariants.toTimestampObject(instance.nextActivation);
-    return jdbc.update(updateWorkflowInstanceSql(), instance.status.name(), instance.state,
+    int updated = jdbc.update(updateWorkflowInstanceSql(), instance.status.name(), instance.state,
         abbreviate(instance.stateText, getInstanceStateTextLength()), nextActivation, nextActivation, nextActivation,
         instance.status == executing ? executorInfo.getExecutorId() : null, instance.retries, instance.businessKey,
         toTimestamp(instance.started), instance.id);
+    if (updated == 0) {
+      logger.warn(
+          "Updating workflow instance {} did not update any rows in the database, instance may have been recovered by another executor.",
```

Comment:
Should we add the same logging to the `updateWorkflowInstanceWithCTE` method too?

Reply:
added